components/raftstore/src/store/peer.rs

```diff
@@ -1827,7 +1850,7 @@ where
 
             if is_read_index_request {
```

Comment:
Is it possible that someone calls `read_index` RPC on a follower and it skips the response checking in `ready_to_handle_unsafe_replica_read`?

Reply:
Good catch.

Reply:
Better to remove this branch? PTAL @NingLin-P